### ISSUE TITLE
"Session upload failed" error when Amazon S3 Object Lock is enabled

### ISSUE THREAD

**Issue by ilpianista** [gravitational/teleport#3300]
**What happened**:

I enabled Amazon S3 Object Lock on my bucket and since then, teleport fails to upload any session.

**What you expected to happen**:

Sessions should be uploaded to the S3 bucket as it was before.

**How to reproduce it (as minimally and precisely as possible)**:

Configure teleport to use S3 as backend. Enable Object Lock on that bucket (from the Properties tab).

**Environment**:
- Teleport version (use `teleport version`): 3.2.6

```
WARN [AUDIT]     "Session upload failed: \nERROR REPORT:\nOriginal Error: *trace.BadParameterError InvalidRequest: Content-MD5 HTTP header is required for Put Object requests with Object Lock parameters\n\tstatus code: 400, request id: 7B0C137DS3949652, host id: +87JKFDJh+k8YH+xBUR/1bXCkKXBH6/uXXpaDOJSxXkhSjeE2idqdfsxoLDSKASAZIMHq9/A=\nStack Trace:\n\t/gopath/src/github.com/gravitational/teleport/lib/events/s3sessions/s3handler.go:247 github.com/gravitational/teleport/lib/events/s3sessions.ConvertS3Error\n\t/gopath/src/github.com/gravitational/teleport/lib/events/s3sessions/s3handler.go:126 github.com/gravitational/teleport/lib/events/s3sessions.(*Handler).Upload\n\t/gopath/src/github.com/gravitational/teleport/lib/events/auditlog.go:310 github.com/gravitational/teleport/lib/events.(*AuditLog).UploadSessionRecording\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/auth_with_roles.go:946 github.com/gravitational/teleport/lib/auth.(*AuthWithRoles).UploadSessionRecording\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/apiserver.go:1868 github.com/gravitational/teleport/lib/auth.(*APIServer).uploadSessionRecording\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/apiserver.go:153 github.com/gravitational/teleport/lib/auth.(*APIServer).uploadSessionRecording-fm\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/apiserver.go:265 github.com/gravitational/teleport/lib/auth.(*APIServer).withAuth.func1\n\t/gopath/src/github.com/gravitational/teleport/lib/httplib/httplib.go:50 github.com/gravitational/teleport/lib/httplib.MakeHandler.func1\n\t/gopath/src/github.com/gravitational/teleport/vendor/github.com/julienschmidt/httprouter/router.go:300 github.com/gravitational/teleport/vendor/github.com/julienschmidt/httprouter.(*Router).ServeHTTP\n\t/gopath/src/github.com/gravitational/teleport/lib/httplib/httplib.go:157 github.com/gravitational/teleport/lib/httplib.RewritePaths.func1\n\t/opt/go/src/net/http/server.go:1965 net/http.HandlerFunc.ServeHTTP\n\t/gopath/src/github.com/gravitational/teleport/lib/auth/middleware.go:294 github.com/gravitational/teleport/lib/auth.(*AuthMiddleware).ServeHTTP\n\t/gopath/src/github.com/gravitational/teleport/vendor/github.com/gravitational/oxy/ratelimit/tokenlimiter.go:119 github.com/gravitational/teleport/vendor/github.com/gravitational/oxy/ratelimit.(*TokenLimiter).ServeHTTP\n\t/gopath/src/github.com/gravitational/teleport/vendor/github.com/gravitational/oxy/connlimit/connlimit.go:72 github.com/gravitational/teleport/vendor/github.com/gravitational/oxy/connlimit.(*ConnLimiter).ServeHTTP\n\t/opt/go/src/net/http/server.go:2742 net/http.serverHandler.ServeHTTP\n\t/opt/go/src/net/http/server.go:1848 net/http.(*conn).serve\n\t/opt/go/src/runtime/asm_amd64.s:1334 runtime.goexit\nUser Message: InvalidRequest: Content-MD5 HTTP header is required for Put Object requests with Object Lock parameters\n\tstatus code: 400, request id: 7B0C137DS3949652, host id: +87JKFDJh+k8YH+xBUR/1bXCkKXBH6/uXXpaDOJSxXkhSjeE2idqdfsxoLDSKASAZIMHq9/A=\n" duration:10.655138ms session-id:b4e4777e-3cf8-11ea-8039-d43d7ed6ca54 events/auditlog.go:313
```

By looking at your Gopkg.lock, it looks like you are using the [AWS SDK v1.12.25](https://github.com/gravitational/teleport/blob/v3.2.6/Gopkg.lock#L96), even in your [latest version](https://github.com/gravitational/teleport/blob/v4.2.2/Gopkg.lock#L126), while this feature has been implemented in the SDK [since 1.13.12](https://github.com/aws/aws-sdk-go/commit/9a78182bdd814365bd31b9c334d515c9d825e322).

Do you support Amazon S3 Object Lock? Your documentation [says so](https://gravitational.com/teleport/docs/aws_oss_guide/#s3).

**Comment by benarent** [comment]
We have a ticket to bump the SDK https://github.com/gravitational/teleport/issues/3210 

```
InvalidRequest: Content-MD5 HTTP header is required for Put Object requests with Object Lock parameter
```
It looks like we might want to add more headers.

**Comment by zmb3** [comment]
Closing due to inactivity

**Comment by diegmonti** [comment]
Notice that this issue was fixed by #3210 because of aws/aws-sdk-go#1827
